### PR TITLE
feat(rust): allow more granular Rust integer types and cleaned up clippy warnings

### PR DESCRIPTION
**Pull request by thrykol**
-  Added CLI option to down cast integers to their best fit
    - Will only occur when `minimum` and `maximum` are both set
    - Configuration defaults to `false`
-  Added CLI option to prefer unsigned integers
    - Will only occur when `minimum` is defined and non-negative
    - Configuration defaults to `false`
-  toParamName now lower cases to adhere to rustfmt
    - Previously, the tail component of `$ref` was being upper cased
-  Added unit tests around all new features

@frol
@farcaller
@richardwhiuk
@paladinzh

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
LGTM. Thanks for the enhancement.

**Comment by spacether** [comment]
Sample up to date test failed in CI for this PR

**Comment by spacether** [comment]
I am fixing it in:
https://github.com/OpenAPITools/openapi-generator/pull/12817

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java [hunk 1]
```diff
@@ -414,7 +456,8 @@ public String toVarName(String name) {
 
     @Override
     public String toParamName(String name) {
-        return toVarName(name);
+        // $ref appears to be all uppercase which is contrary to rustfmt practice so lowercase parameters
+        return toVarName(name.toLowerCase(Locale.ROOT));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java:460]: This result in change of variable name (e.g. pet_id => petid).

I'll file a PR to fix it by remove https://github.com/OpenAPITools/openapi-generator/pull/12479/files#diff-8daa2cc862ff0a63dd72dac551951b1bb07675e3cc0609a6dea55d2e85492520R440 instead.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java:460]: Filed https://github.com/OpenAPITools/openapi-generator/pull/12820

**Comment by wing328** [comment]
Thanks @spacether

**Pull request merged**